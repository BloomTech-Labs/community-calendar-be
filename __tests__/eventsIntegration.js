const {createTestClient} = require('apollo-server-testing');

const {constructTestServer} = require('./__testutils');

const {
    GET_EVENTS,
    GET_EVENT_BY_ID
} = require('./__testQuery.js');


describe('Queries', () => {

  it('fetches events', async () => {
    const {server} = constructTestServer();

    const {query} = createTestClient(server);
    const res = await query({query: GET_EVENTS});

    let events = res.data.events;

    expect(events).toMatchSnapshot();
    
    expect(events).toHaveLength(7);
  });

  it('fetches an event by id', async () => {
    const {server} = constructTestServer();

    const {query} = createTestClient(server);
    const res = await query({query: GET_EVENT_BY_ID, variables: {id: "ck4d7l1bx00ge0784nlj6j3am"}});
    
    let event = res.data.events[0];

    expect(event).toMatchSnapshot();

    expect(event.title).toContain("Art & Architecture - Downtown Walking Tour");


  });
});
