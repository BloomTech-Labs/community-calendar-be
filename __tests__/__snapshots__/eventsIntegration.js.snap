// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Queries fetches an event by id 1`] = `
Object {
  "creator": Object {
    "id": "ck4d7l0vj00ff0784lxympdik",
  },
  "description": "Detroit is a city rich in history, grand buildings, and vibrant art surrounding public spaces. Walk with Detroit Experience Factory through the downtown area as we explore some of the great contributions of both prolific architects and emerging artists.",
  "end": "2019-12-22T19:30:00.000Z",
  "event_images": Array [
    Object {
      "url": "https://img.evbuc.com/https%3A%2F%2Fcdn.evbuc.com%2Fimages%2F69917729%2F89373299245%2F1%2Foriginal.jpg?w=800&auto=format%2Ccompress&q=75&sharp=10&rect=0%2C693%2C4000%2C2000&s=94184073d91c25306d22d1b0401cfea9",
    },
  ],
  "id": "ck4d7l1bx00ge0784nlj6j3am",
  "locations": Array [
    Object {
      "city": "Detroit",
      "id": "ck4d7l1co00gh0784fhkexuea",
      "name": "Roasting Plant",
      "state": "MI",
      "street_address": "600 Woodward Avenue",
      "street_address_2": null,
      "zipcode": 48202,
    },
  ],
  "start": "2019-12-22T17:00:00.000Z",
  "tags": Array [],
  "title": "Art & Architecture - Downtown Walking Tour",
}
`;

exports[`Queries fetches events 1`] = `
Array [
  Object {
    "creator": Object {
      "id": "ck4d7l0vj00ff0784lxympdik",
    },
    "description": "Detroit is a city rich in history, grand buildings, and vibrant art surrounding public spaces. Walk with Detroit Experience Factory through the downtown area as we explore some of the great contributions of both prolific architects and emerging artists.",
    "end": "2019-12-22T19:30:00.000Z",
    "event_images": Array [
      Object {
        "url": "https://img.evbuc.com/https%3A%2F%2Fcdn.evbuc.com%2Fimages%2F69917729%2F89373299245%2F1%2Foriginal.jpg?w=800&auto=format%2Ccompress&q=75&sharp=10&rect=0%2C693%2C4000%2C2000&s=94184073d91c25306d22d1b0401cfea9",
      },
    ],
    "id": "ck4d7l1bx00ge0784nlj6j3am",
    "locations": Array [
      Object {
        "city": "Detroit",
        "id": "ck4d7l1co00gh0784fhkexuea",
        "name": "Roasting Plant",
        "state": "MI",
        "street_address": "600 Woodward Avenue",
        "street_address_2": null,
        "zipcode": 48202,
      },
    ],
    "start": "2019-12-22T17:00:00.000Z",
    "tags": Array [],
    "title": "Art & Architecture - Downtown Walking Tour",
  },
  Object {
    "creator": Object {
      "id": "ck4d7l0vj00ff0784lxympdik",
    },
    "description": "The winter magic isn't over yet - join us for a walk among downtown's gorgeous parks and public spaces, twinkling with lights throughout the winter season. We'll stop to hear from some of the people behind the park lighting.",
    "end": "2020-01-12T02:00:00.000Z",
    "event_images": Array [
      Object {
        "url": "https://img.evbuc.com/https%3A%2F%2Fcdn.evbuc.com%2Fimages%2F85140001%2F89373299245%2F1%2Foriginal.20191219-193513?w=800&auto=format%2Ccompress&q=75&sharp=10&rect=220%2C0%2C1880%2C940&s=5c24d24ce75c0a170c95f533542b323f",
      },
    ],
    "id": "ck4d7l1g500gu0784r68n5rmk",
    "locations": Array [
      Object {
        "city": "Detroit",
        "id": "ck4d7l1gv00gx07844710m9ht",
        "name": "Roasting Plant",
        "state": "MI",
        "street_address": "600 Woodward Avenue",
        "street_address_2": null,
        "zipcode": 48202,
      },
    ],
    "start": "2020-01-12T00:00:00.000Z",
    "tags": Array [],
    "title": "Downtown Winter Wonderland Lights Walk",
  },
  Object {
    "creator": Object {
      "id": "ck4d7l0yb00fk0784o49wkfpw",
    },
    "description": "Join Mary Sheffield and friends for an intimate conversation about purpose, mindset, vision and much more. Let's talk, heal, connect and set our intentions for the 2020 Shift!!",
    "end": "2019-12-28T01:00:00.000Z",
    "event_images": Array [
      Object {
        "url": "https://img.evbuc.com/https%3A%2F%2Fcdn.evbuc.com%2Fimages%2F85046743%2F5763501059%2F1%2Foriginal.20191219-001336?w=800&auto=format%2Ccompress&q=75&sharp=10&rect=0%2C64%2C1080%2C540&s=946121ba79ce53ba2459153c2741b52a",
      },
    ],
    "id": "ck4d7l1kh00ha0784663jxb0t",
    "locations": Array [
      Object {
        "city": "Detroit",
        "id": "ck4d7l1kv00hd078459s6n85p",
        "name": "Trust Cocktails, Shareables & Nightlife",
        "state": "MI",
        "street_address": "205 West Congress Street",
        "street_address_2": null,
        "zipcode": 48226,
      },
    ],
    "start": "2019-12-27T23:00:00.000Z",
    "tags": Array [],
    "title": "2020 Shift Tap In w/ Mary Sheffield",
  },
  Object {
    "creator": Object {
      "id": "ck4d7l11200fp0784xzv9m6au",
    },
    "description": "Another Magical Evening!",
    "end": "2020-01-06T03:00:00.000Z",
    "event_images": Array [
      Object {
        "url": "https://dscclub.com/wp/wp-content/uploads/2014/02/IceShowPage.png",
      },
    ],
    "id": "ck4d7l1ow00hq078482qmhisf",
    "locations": Array [
      Object {
        "city": "Bloomfield Hills",
        "id": "ck4d7l1p800ht0784yzy3uzx1",
        "name": "Detroit Skating Club",
        "state": "MI",
        "street_address": "888 Denison Ct",
        "street_address_2": null,
        "zipcode": 48302,
      },
    ],
    "start": "2020-01-06T00:00:00.000Z",
    "tags": Array [],
    "title": "Detroit Skating Club's 21st Annual Ice Show",
  },
  Object {
    "creator": Object {
      "id": "ck4d7l13z00fu0784srvdlnlg",
    },
    "description": "Kwanzaa is an African American seven-day cultural celebration first celebrated in 1966. Each day honors one of seven principles - the Nguzo Saba. This tour is on Day 1 and highlights African American history connected to the principle of Umoja - Unity.",
    "end": "2019-12-22T19:00:00.000Z",
    "event_images": Array [
      Object {
        "url": "https://img.evbuc.com/https%3A%2F%2Fcdn.evbuc.com%2Fimages%2F83852725%2F88642988607%2F1%2Foriginal.jpg?w=800&auto=format%2Ccompress&q=75&sharp=10&rect=0%2C0%2C800%2C400&s=634757517829d19f5fb6128818867f9f",
      },
    ],
    "id": "ck4d7l1sa00i607846v9id2wk",
    "locations": Array [
      Object {
        "city": "Detroit",
        "id": "ck4d7l1st00i90784b8n7jwdl",
        "name": "Charles H. Wright Museum of African American History",
        "state": "MI",
        "street_address": "315 East Warren Avenue",
        "street_address_2": null,
        "zipcode": 48201,
      },
    ],
    "start": "2019-12-26T17:00:00.000Z",
    "tags": Array [],
    "title": "Kwanzaa Tour at the Charles H. Wright Museum",
  },
  Object {
    "creator": Object {
      "id": "ck4d7l16o00fz0784krzwt7u6",
    },
    "description": "Use TinkerCAD modeling software to create objects that can be 3D printed. 3D printers build up thin layers of plastic to form objects. Free program for youth ages 8-14, Call 313-481-1409 for more information",
    "end": "2020-01-04T20:00:00.000Z",
    "event_images": Array [
      Object {
        "url": "https://img.evbuc.com/https%3A%2F%2Fcdn.evbuc.com%2Fimages%2F84135113%2F293621734690%2F1%2Foriginal.20191210-171159?w=800&auto=format%2Ccompress&q=75&sharp=10&rect=1%2C84%2C224%2C112&s=cf869634d7a67a925a7db1b8ad861f53",
      },
    ],
    "id": "ck4d7l1w900im0784p3r0jr5m",
    "locations": Array [
      Object {
        "city": "Detroit",
        "id": "ck4d7l1wu00ip0784c89pqr46",
        "name": "Detroit Main Library, Children's Room",
        "state": "MI",
        "street_address": "5201 Woodward Avenue",
        "street_address_2": null,
        "zipcode": 48202,
      },
    ],
    "start": "2020-01-04T18:30:00.000Z",
    "tags": Array [],
    "title": "3D Design with TinkerCAD",
  },
  Object {
    "creator": Object {
      "id": "ck4d7l19600g40784gu4maj1p",
    },
    "description": "Proceeds made from tailgating go to support our mission of operating our markets, providing greater access to good food, and growing local food businesses. $45 â€“ $180 depending on location per car",
    "end": "2019-12-30T00:00:00.000Z",
    "event_images": Array [
      Object {
        "url": "https://img.evbuc.com/https%3A%2F%2Fcdn.evbuc.com%2Fimages%2F68668427%2F145956314054%2F1%2Foriginal.20190814-194752?w=800&auto=format%2Ccompress&q=75&sharp=10&rect=0%2C397%2C3456%2C1728&s=1344cbac19465528c33d117ec245f5ba",
      },
    ],
    "id": "ck4d7l20400j20784d0rtdoa7",
    "locations": Array [
      Object {
        "city": "Detroit",
        "id": "ck4d7l20s00j507844cr5qbkk",
        "name": "Eastern Market",
        "state": "MI",
        "street_address": "Eastern Market",
        "street_address_2": null,
        "zipcode": 48208,
      },
    ],
    "start": "2019-12-29T12:00:00.000Z",
    "tags": Array [],
    "title": "Detroit Lions Eastern Market Tailgating",
  },
]
`;
